src/protobuf/message.cr

```diff
@@ -3,10 +3,19 @@ module Protobuf
     macro contract
       FIELDS = {} of Int32 => HashLiteral(Symbol, ASTNode)
       {{yield}}
-      _generate_decoder
-      _generate_encoder
+      _generate_decoder "proto2"
+      _generate_encoder "proto2"
```

Comment:
Good point.  I went with the first option... it took me a bit to figure out the block arguments in macros.